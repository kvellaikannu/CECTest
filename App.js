/**
 * CECTest React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React from 'react';

import {Provider} from 'react-redux';
import {applyMiddleware, createStore} from 'redux';
import logger from 'redux-logger';
import codePush from 'react-native-code-push';
import AppReducer from './reducers/index';

import AppNavigator from './AppNavigator';


const store = createStore(AppReducer, applyMiddleware(logger));

class AppComponent extends React.Component {
    componentDidMount(){
    }

    render() {
        return (
            <Provider store={store}>
                <AppNavigator />
            </Provider>
        );
    }
}

/* Check Update When app Resumes and Install new version upon next restart*/
let codePushOptions = {
    checkFrequency: codePush.CheckFrequency.ON_APP_RESUME,
    InstallMode: codePush.InstallMode.ON_NEXT_RESTART
};

const App = codePush(codePushOptions)(AppComponent);

export default App;