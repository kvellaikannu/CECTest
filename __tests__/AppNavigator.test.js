import 'react-native';
import React from 'react';
import App ,{AppNavigator}from '../AppNavigator';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);
const initialState = {
    nav : {},
    net : {isConnected : false}
};

it('renders Connected AppNavigator as expected', () => {

    const props = {
        navigation : {navigate : jest.fn()}
    };

    let store = mockStore(initialState)
    const wrapper = shallow(
        <App {...props}/>,
        { context: { store: store } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
});