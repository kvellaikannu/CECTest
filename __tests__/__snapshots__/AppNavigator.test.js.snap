// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Connected AppNavigator as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppNavigator
    dispatch={[Function]}
    nav={Object {}}
    navigation={
      Object {
        "navigate": [Function],
      }
    }
    net={
      Object {
        "isConnected": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "navigation": Object {
        "dispatch": [Function],
        "goBack": [Function],
        "navigate": [Function],
        "setParams": [Function],
        "state": Object {},
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "navigation": Object {
          "dispatch": [Function],
          "goBack": [Function],
          "navigate": [Function],
          "setParams": [Function],
          "state": Object {},
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
